From 41dc89521e329bccedc0823603e8b9916b9cc7b4 Mon Sep 17 00:00:00 2001
From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Mon, 1 Jul 2019 21:52:07 +0200
Subject: [PATCH 1/5] luci-base: move some generic classes into a separate
 luci-base-libs package

The new package luci-base-libs provides the modules that not strictly relate
to the web-interface of luci. By separating these libs they can be used by
other packages without having to install the web-components.
This change was inspired by providing a shell-only interface for 4MB-flash
devices, by keeping as much code common with a full install.

Signed-off-by: Sven Roederer <freifunk@it-solutions.geroedel.de>
---
 libs/luci-lib-base/Makefile                        | 14 ++++++++++++++
 .../luci-lib-base}/luasrc/debug.lua                |  0
 .../luci-lib-base}/luasrc/http.lua                 |  0
 .../luci-lib-base}/luasrc/http.luadoc              |  0
 .../luci-lib-base}/luasrc/ltn12.lua                |  0
 .../luci-lib-base}/luasrc/util.lua                 |  0
 .../luci-lib-base}/luasrc/util.luadoc              |  0
 libs/luci-lib-httpclient/Makefile                  |  2 +-
 libs/luci-lib-httpprotoutils/Makefile              |  2 +-
 9 files changed, 16 insertions(+), 2 deletions(-)
 create mode 100644 libs/luci-lib-base/Makefile
 rename {modules/luci-base => libs/luci-lib-base}/luasrc/debug.lua (100%)
 rename {modules/luci-base => libs/luci-lib-base}/luasrc/http.lua (100%)
 rename {modules/luci-base => libs/luci-lib-base}/luasrc/http.luadoc (100%)
 rename {modules/luci-base => libs/luci-lib-base}/luasrc/ltn12.lua (100%)
 rename {modules/luci-base => libs/luci-lib-base}/luasrc/util.lua (100%)
 rename {modules/luci-base => libs/luci-lib-base}/luasrc/util.luadoc (100%)

diff --git a/libs/luci-lib-base/Makefile b/libs/luci-lib-base/Makefile
new file mode 100644
index 000000000..35b1836ec
--- /dev/null
+++ b/libs/luci-lib-base/Makefile
@@ -0,0 +1,14 @@
+#
+# Copyright (C) 2008-2014 The LuCI Team <luci@lists.subsignal.org>
+#
+# This is free software, licensed under the Apache License, Version 2.0 .
+#
+
+include $(TOPDIR)/rules.mk
+
+LUCI_TITLE:=basic libraries for luci
+LUCI_DEPENDS:=+lua +luci-lib-nixio +luci-lib-ip +luci-lib-jsonc +liblucihttp-lua
+
+include ../../luci.mk
+
+# call BuildPackage - OpenWrt buildroot signature
diff --git a/modules/luci-base/luasrc/debug.lua b/libs/luci-lib-base/luasrc/debug.lua
similarity index 100%
rename from modules/luci-base/luasrc/debug.lua
rename to libs/luci-lib-base/luasrc/debug.lua
diff --git a/modules/luci-base/luasrc/http.lua b/libs/luci-lib-base/luasrc/http.lua
similarity index 100%
rename from modules/luci-base/luasrc/http.lua
rename to libs/luci-lib-base/luasrc/http.lua
diff --git a/modules/luci-base/luasrc/http.luadoc b/libs/luci-lib-base/luasrc/http.luadoc
similarity index 100%
rename from modules/luci-base/luasrc/http.luadoc
rename to libs/luci-lib-base/luasrc/http.luadoc
diff --git a/modules/luci-base/luasrc/ltn12.lua b/libs/luci-lib-base/luasrc/ltn12.lua
similarity index 100%
rename from modules/luci-base/luasrc/ltn12.lua
rename to libs/luci-lib-base/luasrc/ltn12.lua
diff --git a/modules/luci-base/luasrc/util.lua b/libs/luci-lib-base/luasrc/util.lua
similarity index 100%
rename from modules/luci-base/luasrc/util.lua
rename to libs/luci-lib-base/luasrc/util.lua
diff --git a/modules/luci-base/luasrc/util.luadoc b/libs/luci-lib-base/luasrc/util.luadoc
similarity index 100%
rename from modules/luci-base/luasrc/util.luadoc
rename to libs/luci-lib-base/luasrc/util.luadoc
diff --git a/libs/luci-lib-httpclient/Makefile b/libs/luci-lib-httpclient/Makefile
index b8bd428b2..9c28c35d8 100644
--- a/libs/luci-lib-httpclient/Makefile
+++ b/libs/luci-lib-httpclient/Makefile
@@ -7,7 +7,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=HTTP(S) client library
-LUCI_DEPENDS:=+luci-base +luci-lib-nixio +luci-lib-httpprotoutils
+LUCI_DEPENDS:=+luci-lib-base +luci-lib-nixio +luci-lib-httpprotoutils
 
 include ../../luci.mk
 
diff --git a/libs/luci-lib-httpprotoutils/Makefile b/libs/luci-lib-httpprotoutils/Makefile
index 851a362eb..95f45d200 100644
--- a/libs/luci-lib-httpprotoutils/Makefile
+++ b/libs/luci-lib-httpprotoutils/Makefile
@@ -7,7 +7,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=HTTP protocol utility functions
-LUCI_DEPENDS:=+luci-base
+LUCI_DEPENDS:=+luci-lib-base
 
 include ../../luci.mk
 
-- 
2.20.1


From 302bd2d33d825d51814c0e9984c6fa4bf577d762 Mon Sep 17 00:00:00 2001
From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Sat, 6 Jul 2019 14:18:15 +0200
Subject: [PATCH 2/5] luci-base(-libs): move pcdata() and striptags() from
 util- to xml-class

To complete the previous commit these functions are defined in the resulting
luci-base package but are also used in the new luci-base-libs package. So
move them into the new xml-module of the new package.

Signed-off-by: Sven Roederer <freifunk@it-solutions.geroedel.de>
---
 libs/luci-lib-base/luasrc/util.lua    | 10 ----------
 libs/luci-lib-base/luasrc/util.luadoc | 18 ------------------
 modules/luci-base/luasrc/xml.lua      | 26 ++++++++++++++++++++++++++
 modules/luci-base/luasrc/xml.luadoc   | 23 +++++++++++++++++++++++
 4 files changed, 49 insertions(+), 28 deletions(-)
 create mode 100644 modules/luci-base/luasrc/xml.lua
 create mode 100644 modules/luci-base/luasrc/xml.luadoc

diff --git a/libs/luci-lib-base/luasrc/util.lua b/libs/luci-lib-base/luasrc/util.lua
index a30e8b72f..f3aa52603 100644
--- a/libs/luci-lib-base/luasrc/util.lua
+++ b/libs/luci-lib-base/luasrc/util.lua
@@ -159,10 +159,6 @@ end
 -- String and data manipulation routines
 --
 
-function pcdata(value)
-	return value and tparser.pcdata(tostring(value))
-end
-
 function urlencode(value)
 	if value ~= nil then
 		local str = tostring(value)
@@ -182,10 +178,6 @@ function urldecode(value, decode_plus)
 	return nil
 end
 
-function striptags(value)
-	return value and tparser.striptags(tostring(value))
-end
-
 function shellquote(value)
 	return string.format("'%s'", string.gsub(value or "", "'", "'\\''"))
 end
@@ -343,8 +335,6 @@ function parse_units(ustr)
 end
 
 -- also register functions above in the central string class for convenience
-string.pcdata      = pcdata
-string.striptags   = striptags
 string.split       = split
 string.trim        = trim
 string.cmatch      = cmatch
diff --git a/libs/luci-lib-base/luasrc/util.luadoc b/libs/luci-lib-base/luasrc/util.luadoc
index 4ec68dd1e..6ab130008 100644
--- a/libs/luci-lib-base/luasrc/util.luadoc
+++ b/libs/luci-lib-base/luasrc/util.luadoc
@@ -67,15 +67,6 @@ Recursively dumps a table to stdout, useful for testing and debugging.
 @return				Always nil
 ]]
 
----[[
-Create valid XML PCDATA from given string.
-
-@class				function
-@name				pcdata
-@param value		String value containing the data to escape
-@return				String value containing the escaped data
-]]
-
 ---[[
 Decode an URL-encoded string - optionally decoding the "+" sign to space.
 
@@ -97,15 +88,6 @@ URL-encode given string.
 @see				urldecode
 ]]
 
----[[
-Strip HTML tags from given string.
-
-@class				function
-@name				striptags
-@param value		String containing the HTML text
-@return				String with HTML tags stripped of
-]]
-
 ---[[
 Safely quote value for use in shell commands.
 
diff --git a/modules/luci-base/luasrc/xml.lua b/modules/luci-base/luasrc/xml.lua
new file mode 100644
index 000000000..30b37210b
--- /dev/null
+++ b/modules/luci-base/luasrc/xml.lua
@@ -0,0 +1,26 @@
+-- Copyright 2008 Steven Barth <steven@midlink.org>
+-- Licensed to the public under the Apache License 2.0.
+
+local tparser = require "luci.template.parser"
+local string = require "string"
+
+local tostring = tostring
+
+module "luci.xml"
+
+--
+-- String and data manipulation routines
+--
+
+function pcdata(value)
+	return value and tparser.pcdata(tostring(value))
+end
+
+function striptags(value)
+	return value and tparser.striptags(tostring(value))
+end
+
+
+-- also register functions above in the central string class for convenience
+string.pcdata      = pcdata
+string.striptags   = striptags
diff --git a/modules/luci-base/luasrc/xml.luadoc b/modules/luci-base/luasrc/xml.luadoc
new file mode 100644
index 000000000..58de53396
--- /dev/null
+++ b/modules/luci-base/luasrc/xml.luadoc
@@ -0,0 +1,23 @@
+---[[
+LuCI utility functions.
+]]
+module "luci.xml"
+
+---[[
+Create valid XML PCDATA from given string.
+
+@class				function
+@name				pcdata
+@param value		String value containing the data to escape
+@return				String value containing the escaped data
+]]
+
+---[[
+Strip HTML tags from given string.
+
+@class				function
+@name				striptags
+@param value		String containing the HTML text
+@return				String with HTML tags stripped of
+]]
+
-- 
2.20.1


From 454a566a4fe4924ad3f326a7f63fe06daefca7f0 Mon Sep 17 00:00:00 2001
From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Sun, 29 Mar 2020 14:00:49 +0200
Subject: [PATCH 3/5] docs: move pcdata() and striptags() from util- to
 xml-class

update according to previous commit

Signed-off-by: Sven Roederer <freifunk@it-solutions.geroedel.de>
---
 docs/api/index.html                          |   9 +
 docs/api/modules/luci.dispatcher.html        |   4 +
 docs/api/modules/luci.http.conditionals.html |   4 +
 docs/api/modules/luci.http.date.html         |   4 +
 docs/api/modules/luci.http.html              |   4 +
 docs/api/modules/luci.http.mime.html         |   4 +
 docs/api/modules/luci.i18n.html              |   4 +
 docs/api/modules/luci.ip.cidr.html           |   4 +
 docs/api/modules/luci.ip.html                |   4 +
 docs/api/modules/luci.json.html              |   4 +
 docs/api/modules/luci.jsonc.html             |   4 +
 docs/api/modules/luci.jsonc.parser.html      |   4 +
 docs/api/modules/luci.model.ipkg.html        |   4 +
 docs/api/modules/luci.model.uci.html         |   4 +
 docs/api/modules/luci.rpcc.html              |   4 +
 docs/api/modules/luci.rpcc.ruci.html         |   4 +
 docs/api/modules/luci.sys.html               |   4 +
 docs/api/modules/luci.sys.init.html          |   4 +
 docs/api/modules/luci.sys.iptparser.html     |   4 +
 docs/api/modules/luci.sys.net.html           |   4 +
 docs/api/modules/luci.sys.process.html       |   4 +
 docs/api/modules/luci.sys.user.html          |   4 +
 docs/api/modules/luci.sys.wifi.html          |   4 +
 docs/api/modules/luci.util.html              |  82 +----
 docs/api/modules/luci.xml.html               | 321 +++++++++++++++++++
 docs/api/modules/nixio.CHANGELOG.html        |   4 +
 docs/api/modules/nixio.CryptoHash.html       |   4 +
 docs/api/modules/nixio.File.html             |   4 +
 docs/api/modules/nixio.README.html           |   4 +
 docs/api/modules/nixio.Socket.html           |   4 +
 docs/api/modules/nixio.TLSContext.html       |   4 +
 docs/api/modules/nixio.TLSSocket.html        |   4 +
 docs/api/modules/nixio.UnifiedIO.html        |   4 +
 docs/api/modules/nixio.bin.html              |   4 +
 docs/api/modules/nixio.bit.html              |   4 +
 docs/api/modules/nixio.crypto.html           |   4 +
 docs/api/modules/nixio.fs.html               |   4 +
 docs/api/modules/nixio.html                  |   4 +
 38 files changed, 474 insertions(+), 78 deletions(-)
 create mode 100644 docs/api/modules/luci.xml.html

diff --git a/docs/api/index.html b/docs/api/index.html
index 5e3f3c211..66a27a33b 100644
--- a/docs/api/index.html
+++ b/docs/api/index.html
@@ -126,6 +126,10 @@
 		<a href="modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="modules/nixio.html">nixio</a>
 	</li>
@@ -328,6 +332,11 @@ LuCI system utilities / wifi related functions.</td>
 		<td class="summary"></td>
 	</tr>
 
+	<tr>
+		<td class="name"><a href="modules/luci.xml.html">luci.xml</a></td>
+		<td class="summary"></td>
+	</tr>
+
 	<tr>
 		<td class="name"><a href="modules/nixio.html">nixio</a></td>
 		<td class="summary">
diff --git a/docs/api/modules/luci.dispatcher.html b/docs/api/modules/luci.dispatcher.html
index ea33a5c70..32cb49573 100644
--- a/docs/api/modules/luci.dispatcher.html
+++ b/docs/api/modules/luci.dispatcher.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.http.conditionals.html b/docs/api/modules/luci.http.conditionals.html
index 8c940bd3f..14ee627f6 100644
--- a/docs/api/modules/luci.http.conditionals.html
+++ b/docs/api/modules/luci.http.conditionals.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.http.date.html b/docs/api/modules/luci.http.date.html
index 1ec5beb8b..680dd4313 100644
--- a/docs/api/modules/luci.http.date.html
+++ b/docs/api/modules/luci.http.date.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.http.html b/docs/api/modules/luci.http.html
index 473172784..3e8799e74 100644
--- a/docs/api/modules/luci.http.html
+++ b/docs/api/modules/luci.http.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.http.mime.html b/docs/api/modules/luci.http.mime.html
index 85eaf53ab..756b0727a 100644
--- a/docs/api/modules/luci.http.mime.html
+++ b/docs/api/modules/luci.http.mime.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.i18n.html b/docs/api/modules/luci.i18n.html
index 0f315bebd..5ba44b4c0 100644
--- a/docs/api/modules/luci.i18n.html
+++ b/docs/api/modules/luci.i18n.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.ip.cidr.html b/docs/api/modules/luci.ip.cidr.html
index ae6c61dc5..cfe190935 100644
--- a/docs/api/modules/luci.ip.cidr.html
+++ b/docs/api/modules/luci.ip.cidr.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.ip.html b/docs/api/modules/luci.ip.html
index 0599396b6..8d503bd65 100644
--- a/docs/api/modules/luci.ip.html
+++ b/docs/api/modules/luci.ip.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.json.html b/docs/api/modules/luci.json.html
index db2d1da3f..5885556d5 100644
--- a/docs/api/modules/luci.json.html
+++ b/docs/api/modules/luci.json.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.jsonc.html b/docs/api/modules/luci.jsonc.html
index 79deb933d..f6ab990fa 100644
--- a/docs/api/modules/luci.jsonc.html
+++ b/docs/api/modules/luci.jsonc.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.jsonc.parser.html b/docs/api/modules/luci.jsonc.parser.html
index 709cb9afc..54e1e487d 100644
--- a/docs/api/modules/luci.jsonc.parser.html
+++ b/docs/api/modules/luci.jsonc.parser.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.model.ipkg.html b/docs/api/modules/luci.model.ipkg.html
index a0af3187b..e97097db5 100644
--- a/docs/api/modules/luci.model.ipkg.html
+++ b/docs/api/modules/luci.model.ipkg.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.model.uci.html b/docs/api/modules/luci.model.uci.html
index c1eaf5f81..84b6ecc1b 100644
--- a/docs/api/modules/luci.model.uci.html
+++ b/docs/api/modules/luci.model.uci.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.rpcc.html b/docs/api/modules/luci.rpcc.html
index 18065788b..10d628ff4 100644
--- a/docs/api/modules/luci.rpcc.html
+++ b/docs/api/modules/luci.rpcc.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.rpcc.ruci.html b/docs/api/modules/luci.rpcc.ruci.html
index 7348f1932..598020df7 100644
--- a/docs/api/modules/luci.rpcc.ruci.html
+++ b/docs/api/modules/luci.rpcc.ruci.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.html b/docs/api/modules/luci.sys.html
index b86d280fb..f601e17dd 100644
--- a/docs/api/modules/luci.sys.html
+++ b/docs/api/modules/luci.sys.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.init.html b/docs/api/modules/luci.sys.init.html
index e2c51f953..b178cafca 100644
--- a/docs/api/modules/luci.sys.init.html
+++ b/docs/api/modules/luci.sys.init.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.iptparser.html b/docs/api/modules/luci.sys.iptparser.html
index 5928281cf..aa9f7ce1c 100644
--- a/docs/api/modules/luci.sys.iptparser.html
+++ b/docs/api/modules/luci.sys.iptparser.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.net.html b/docs/api/modules/luci.sys.net.html
index e7802bb24..bf1bb54aa 100644
--- a/docs/api/modules/luci.sys.net.html
+++ b/docs/api/modules/luci.sys.net.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.process.html b/docs/api/modules/luci.sys.process.html
index d3664817c..ec42ecadd 100644
--- a/docs/api/modules/luci.sys.process.html
+++ b/docs/api/modules/luci.sys.process.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.user.html b/docs/api/modules/luci.sys.user.html
index b2307ad38..27b13ccbb 100644
--- a/docs/api/modules/luci.sys.user.html
+++ b/docs/api/modules/luci.sys.user.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.sys.wifi.html b/docs/api/modules/luci.sys.wifi.html
index 6c893244c..37e7053fd 100644
--- a/docs/api/modules/luci.sys.wifi.html
+++ b/docs/api/modules/luci.sys.wifi.html
@@ -124,6 +124,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/luci.util.html b/docs/api/modules/luci.util.html
index f8baddf41..a8f8737aa 100644
--- a/docs/api/modules/luci.util.html
+++ b/docs/api/modules/luci.util.html
@@ -124,6 +124,10 @@
 
 	<li><strong>luci.util</strong></li>
 	
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
@@ -341,13 +345,6 @@ Parse certain units from the given string and return the canonical integer
 value or 0 if the unit is unknown.</td>
 	</tr>
 
-	<tr>
-	<td class="name" nowrap><a href="#pcdata">pcdata</a>&nbsp;(value)</td>
-	<td class="summary">
- 
-Create valid XML PCDATA from given string.</td>
-	</tr>
-
 	<tr>
 	<td class="name" nowrap><a href="#perror">perror</a>&nbsp;(obj)</td>
 	<td class="summary">
@@ -408,13 +405,6 @@ containing the resulting substrings.</td>
 Strips unnecessary lua bytecode from given string.</td>
 	</tr>
 
-	<tr>
-	<td class="name" nowrap><a href="#striptags">striptags</a>&nbsp;(value)</td>
-	<td class="summary">
- 
-Strip HTML tags from given string.</td>
-	</tr>
-
 	<tr>
 	<td class="name" nowrap><a href="#threadlocal">threadlocal</a>&nbsp;()</td>
 	<td class="summary">
@@ -1178,38 +1168,6 @@ Number containing the canonical value
 
 
 
-<dt><a name="pcdata"></a><strong>pcdata</strong>&nbsp;(value)</dt>
-<dd>
-
- 
-Create valid XML PCDATA from given string. 
-
-
-
-<h3>Parameters</h3>
-<ul>
-	
-	<li>
-	  value: String value containing the data to escape
-	</li>
-	
-</ul>
-
-
-
-
-
-
-<h3>Return value:</h3>
-String value containing the escaped data
-
-
-
-</dd>
-
-
-
-
 <dt><a name="perror"></a><strong>perror</strong>&nbsp;(obj)</dt>
 <dd>
 
@@ -1524,38 +1482,6 @@ String value containing the stripped lua byte code
 
 
 
-<dt><a name="striptags"></a><strong>striptags</strong>&nbsp;(value)</dt>
-<dd>
-
- 
-Strip HTML tags from given string. 
-
-
-
-<h3>Parameters</h3>
-<ul>
-	
-	<li>
-	  value: String containing the HTML text
-	</li>
-	
-</ul>
-
-
-
-
-
-
-<h3>Return value:</h3>
-String with HTML tags stripped of
-
-
-
-</dd>
-
-
-
-
 <dt><a name="threadlocal"></a><strong>threadlocal</strong>&nbsp;()</dt>
 <dd>
 
diff --git a/docs/api/modules/luci.xml.html b/docs/api/modules/luci.xml.html
new file mode 100644
index 000000000..63a91e4eb
--- /dev/null
+++ b/docs/api/modules/luci.xml.html
@@ -0,0 +1,321 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
+   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html>
+<head>
+    <title>Reference</title>
+    <link rel="stylesheet" href="../luadoc.css" type="text/css" />
+	<!--meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/-->
+</head>
+
+<body>
+<div id="container">
+
+<div id="product">
+	<div id="product_logo"></div>
+	<div id="product_name"><big><b></b></big></div>
+	<div id="product_description"></div>
+</div> <!-- id="product" -->
+
+<div id="main">
+
+<div id="navigation">
+
+
+<h1>LuaDoc</h1>
+<ul>
+	
+	<li><a href="../index.html">Index</a></li>
+	
+</ul>
+
+
+<!-- Module list -->
+
+<h1>Modules</h1>
+<ul>
+
+	<li>
+		<a href="../modules/luci.dispatcher.html">luci.dispatcher</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.http.html">luci.http</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.http.conditionals.html">luci.http.conditionals</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.http.date.html">luci.http.date</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.http.mime.html">luci.http.mime</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.i18n.html">luci.i18n</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.ip.html">luci.ip</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.ip.cidr.html">luci.ip.cidr</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.json.html">luci.json</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.jsonc.html">luci.jsonc</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.jsonc.parser.html">luci.jsonc.parser</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.model.ipkg.html">luci.model.ipkg</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.model.uci.html">luci.model.uci</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.rpcc.html">luci.rpcc</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.rpcc.ruci.html">luci.rpcc.ruci</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.html">luci.sys</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.init.html">luci.sys.init</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.iptparser.html">luci.sys.iptparser</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.net.html">luci.sys.net</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.process.html">luci.sys.process</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.user.html">luci.sys.user</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.sys.wifi.html">luci.sys.wifi</a>
+	</li>
+
+	<li>
+		<a href="../modules/luci.util.html">luci.sys.wifi</a>
+	</li>
+
+	<li><strong>luci.xml</strong></li>
+	
+	<li>
+		<a href="../modules/nixio.html">nixio</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.CHANGELOG.html">nixio.CHANGELOG</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.CryptoHash.html">nixio.CryptoHash</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.File.html">nixio.File</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.README.html">nixio.README</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.Socket.html">nixio.Socket</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.TLSContext.html">nixio.TLSContext</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.TLSSocket.html">nixio.TLSSocket</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.UnifiedIO.html">nixio.UnifiedIO</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.bin.html">nixio.bin</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.bit.html">nixio.bit</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.crypto.html">nixio.crypto</a>
+	</li>
+
+	<li>
+		<a href="../modules/nixio.fs.html">nixio.fs</a>
+	</li>
+
+</ul>
+
+
+
+<!-- File list -->
+
+
+
+
+
+
+
+</div><!-- id="navigation" -->
+
+<div id="content">
+
+<h1>Class <code>luci.xml</code></h1>
+
+<p></p>
+
+
+
+
+
+
+
+<h2>Functions</h2>
+<table class="function_list">
+
+	<tr>
+	<td class="name" nowrap><a href="#pcdata">pcdata</a>&nbsp;(value)</td>
+	<td class="summary">
+ 
+Create valid XML PCDATA from given string.</td>
+	</tr>
+
+	<tr>
+	<td class="name" nowrap><a href="#striptags">striptags</a>&nbsp;(value)</td>
+	<td class="summary">
+ 
+Strip HTML tags from given string.</td>
+	</tr>
+
+</table>
+
+
+
+
+
+
+<br/>
+<br/>
+
+
+<h2><a name="functions"></a>Functions</h2>
+<dl class="function">
+
+
+
+<dt><a name="pcdata"></a><strong>pcdata</strong>&nbsp;(value)</dt>
+<dd>
+
+ 
+Create valid XML PCDATA from given string. 
+
+
+
+<h3>Parameters</h3>
+<ul>
+	
+	<li>
+	  value: String value containing the data to escape
+	</li>
+	
+</ul>
+
+
+
+
+
+
+<h3>Return value:</h3>
+String value containing the escaped data
+
+
+
+</dd>
+
+
+
+
+<dt><a name="striptags"></a><strong>striptags</strong>&nbsp;(value)</dt>
+<dd>
+
+ 
+Strip HTML tags from given string. 
+
+
+
+<h3>Parameters</h3>
+<ul>
+	
+	<li>
+	  value: String containing the HTML text
+	</li>
+	
+</ul>
+
+
+
+
+
+
+<h3>Return value:</h3>
+String with HTML tags stripped of
+
+
+
+</dd>
+
+
+
+</dl>
+
+
+
+
+
+</div> <!-- id="content" -->
+
+</div> <!-- id="main" -->
+
+<div id="about">
+	<p><a href="http://validator.w3.org/check?uri=referer"><img src="http://www.w3.org/Icons/valid-xhtml10" alt="Valid XHTML 1.0!" height="31" width="88" /></a></p>
+</div> <!-- id="about" -->
+
+</div> <!-- id="container" -->
+</body>
+</html>
diff --git a/docs/api/modules/nixio.CHANGELOG.html b/docs/api/modules/nixio.CHANGELOG.html
index 94b44a5ca..0def9a451 100644
--- a/docs/api/modules/nixio.CHANGELOG.html
+++ b/docs/api/modules/nixio.CHANGELOG.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.CryptoHash.html b/docs/api/modules/nixio.CryptoHash.html
index 7d2f48b1e..a333c407d 100644
--- a/docs/api/modules/nixio.CryptoHash.html
+++ b/docs/api/modules/nixio.CryptoHash.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.File.html b/docs/api/modules/nixio.File.html
index 7a7500a77..b46a317bc 100644
--- a/docs/api/modules/nixio.File.html
+++ b/docs/api/modules/nixio.File.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.README.html b/docs/api/modules/nixio.README.html
index e140659cc..b8481ef55 100644
--- a/docs/api/modules/nixio.README.html
+++ b/docs/api/modules/nixio.README.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.Socket.html b/docs/api/modules/nixio.Socket.html
index 185099125..f1d48ef2b 100644
--- a/docs/api/modules/nixio.Socket.html
+++ b/docs/api/modules/nixio.Socket.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.TLSContext.html b/docs/api/modules/nixio.TLSContext.html
index c84d31895..247670c56 100644
--- a/docs/api/modules/nixio.TLSContext.html
+++ b/docs/api/modules/nixio.TLSContext.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.TLSSocket.html b/docs/api/modules/nixio.TLSSocket.html
index 5d6098a9f..20cdaa55a 100644
--- a/docs/api/modules/nixio.TLSSocket.html
+++ b/docs/api/modules/nixio.TLSSocket.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.UnifiedIO.html b/docs/api/modules/nixio.UnifiedIO.html
index 6410ffb42..8521e3e2a 100644
--- a/docs/api/modules/nixio.UnifiedIO.html
+++ b/docs/api/modules/nixio.UnifiedIO.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.bin.html b/docs/api/modules/nixio.bin.html
index c8ef01831..aeb14eb53 100644
--- a/docs/api/modules/nixio.bin.html
+++ b/docs/api/modules/nixio.bin.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.bit.html b/docs/api/modules/nixio.bit.html
index 419736363..d0ce111db 100644
--- a/docs/api/modules/nixio.bit.html
+++ b/docs/api/modules/nixio.bit.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.crypto.html b/docs/api/modules/nixio.crypto.html
index b91fe28c1..498620bcd 100644
--- a/docs/api/modules/nixio.crypto.html
+++ b/docs/api/modules/nixio.crypto.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.fs.html b/docs/api/modules/nixio.fs.html
index c9d34591c..956c0e93a 100644
--- a/docs/api/modules/nixio.fs.html
+++ b/docs/api/modules/nixio.fs.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li>
 		<a href="../modules/nixio.html">nixio</a>
 	</li>
diff --git a/docs/api/modules/nixio.html b/docs/api/modules/nixio.html
index bf93619c9..36895486c 100644
--- a/docs/api/modules/nixio.html
+++ b/docs/api/modules/nixio.html
@@ -126,6 +126,10 @@
 		<a href="../modules/luci.util.html">luci.util</a>
 	</li>
 
+	<li>
+		<a href="../modules/luci.xml.html">luci.xml</a>
+	</li>
+
 	<li><strong>nixio</strong></li>
 	
 	<li>
-- 
2.20.1


From 67eb84b856054cae00c44bd579e7f208e1a517a8 Mon Sep 17 00:00:00 2001
From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Sat, 6 Jul 2019 14:27:42 +0200
Subject: [PATCH 4/5] treewide: adapt to new luci.xml class

In teh previous commit the luci.xml module was created, Let'S change all
references to the old functions to the new xml-module.

Signed-off-by: Sven Roederer <freifunk@it-solutions.geroedel.de>
---
 .../luasrc/view/openvpn/cbi-select-input-add.htm    |  2 +-
 .../luasrc/view/tinyproxy_status.htm                |  4 ++--
 .../luasrc/view/travelmate/wifi_scan.htm            | 13 +++++++------
 modules/luci-base/luasrc/dispatcher.lua             |  9 +++++----
 .../luasrc/model/cbi/admin_system/backupfiles.lua   |  2 +-
 5 files changed, 16 insertions(+), 14 deletions(-)

diff --git a/applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm b/applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm
index 9ca1e87fa..6f3263590 100644
--- a/applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm
+++ b/applications/luci-app-openvpn/luasrc/view/openvpn/cbi-select-input-add.htm
@@ -84,7 +84,7 @@
 				<select id="instance_template" name="cbi.cts.<%=self.config%>.<%=self.sectiontype%>.select">
 					<option value="" selected="selected" disabled="disabled"><%:Select template ...%></option>
 					<%- for k, v in luci.util.kspairs(self.add_select_options) do %>
-						<option value="<%=k%>"><%=luci.util.pcdata(v)%></option>
+						<option value="<%=k%>"><%=luci.xml.pcdata(v)%></option>
 					<% end -%>
 				</select>
 			</div>
diff --git a/applications/luci-app-tinyproxy/luasrc/view/tinyproxy_status.htm b/applications/luci-app-tinyproxy/luasrc/view/tinyproxy_status.htm
index 2ba9dddb8..9ff0b0292 100644
--- a/applications/luci-app-tinyproxy/luasrc/view/tinyproxy_status.htm
+++ b/applications/luci-app-tinyproxy/luasrc/view/tinyproxy_status.htm
@@ -35,8 +35,8 @@ if luci.http.formvalue("frame") == "1" then
 	if not data then
 		luci.http.write(translate("Failed to retrieve statistics from url:"))
 		luci.http.write(" http://%s:%s" %{
-			luci.util.pcdata(addr),
-			luci.util.pcdata(port)
+			luci.xml.pcdata(addr),
+			luci.xml.pcdata(port)
 		})
 	end
 
diff --git a/applications/luci-app-travelmate/luasrc/view/travelmate/wifi_scan.htm b/applications/luci-app-travelmate/luasrc/view/travelmate/wifi_scan.htm
index 7d00eceed..491af1d92 100644
--- a/applications/luci-app-travelmate/luasrc/view/travelmate/wifi_scan.htm
+++ b/applications/luci-app-travelmate/luasrc/view/travelmate/wifi_scan.htm
@@ -6,6 +6,7 @@ This is free software, licensed under the Apache License, Version 2.0
 <%-
 	local sys = require("luci.sys")
 	local utl = require("luci.util")
+	local xml = require("luci.xml")
 	local dev = luci.http.formvalue("device")
 	local ifn = utl.trim(sys.exec("/bin/ubus -S call network.wireless status 2>/dev/null | jsonfilter -l1 -e '@." .. dev .. ".interfaces[@.config.mode=\"sta\"].ifname' 2>/dev/null"))
 	local iw
@@ -47,10 +48,10 @@ This is free software, licensed under the Apache License, Version 2.0
 			<%- for i, net in ipairs(iw.scanlist or { }) do -%>
 			<div class="tr cbi-section-table-row cbi-rowstyle-1">
 				<div class="td left" style="text-align: left !important">
-					<%=net.ssid and utl.pcdata(net.ssid) or "<em>%s</em>" % translate("hidden")%>
+					<%=net.ssid and xml.pcdata(net.ssid) or "<em>%s</em>" % translate("hidden")%>
 				</div>
 				<div class="td left" style="text-align: left !important">
-					<%=net.bssid and utl.pcdata(net.bssid)%>
+					<%=net.bssid and xml.pcdata(net.bssid)%>
 				</div>
 				<div class="td left" style="text-align: left !important">
 					<%=net.encryption.description%>
@@ -61,9 +62,9 @@ This is free software, licensed under the Apache License, Version 2.0
 				<div class="td cbi-section-actions">
 					<form class="inline" action="<%=luci.dispatcher.build_url('admin/services/travelmate/wifiadd')%>" method="post">
 						<input type="hidden" name="token" value="<%=token%>"/>
-						<input type="hidden" name="device" value="<%=utl.pcdata(dev)%>"/>
-						<input type="hidden" name="ssid" value="<%=utl.pcdata(net.ssid)%>"/>
-						<input type="hidden" name="bssid" value="<%=utl.pcdata(net.bssid)%>"/>
+						<input type="hidden" name="device" value="<%=xml.pcdata(dev)%>"/>
+						<input type="hidden" name="ssid" value="<%=xml.pcdata(net.ssid)%>"/>
+						<input type="hidden" name="bssid" value="<%=xml.pcdata(net.bssid)%>"/>
 						<input type="hidden" name="description" value="<%=net.encryption.description%>"/>
 						<input type="hidden" name="wep" value="<%=net.encryption.wep and 1 or 0%>"/>
 						<%- if net.encryption.wpa then -%>
@@ -85,7 +86,7 @@ This is free software, licensed under the Apache License, Version 2.0
 		</form>
 		<form class="inline" action="<%=luci.dispatcher.build_url('admin/services/travelmate/wifiscan')%>" method="post">
 			<input type="hidden" name="token" value="<%=token%>"/>
-			<input type="hidden" name="device" value="<%=utl.pcdata(dev)%>"/>
+			<input type="hidden" name="device" value="<%=xml.pcdata(dev)%>"/>
 			<input class="cbi-button cbi-input-find" type="submit" value="<%:Repeat scan%>"/>
 		</form>
 	</div>
diff --git a/modules/luci-base/luasrc/dispatcher.lua b/modules/luci-base/luasrc/dispatcher.lua
index d14a86673..79ccf5892 100644
--- a/modules/luci-base/luasrc/dispatcher.lua
+++ b/modules/luci-base/luasrc/dispatcher.lua
@@ -5,6 +5,7 @@
 local fs = require "nixio.fs"
 local sys = require "luci.sys"
 local util = require "luci.util"
+local xml = require "luci.xml"
 local http = require "luci.http"
 local nixio = require "nixio", require "nixio.util"
 
@@ -276,7 +277,7 @@ local function tree_to_json(node, json)
 	if type(node.nodes) == "table" then
 		for subname, subnode in pairs(node.nodes) do
 			local spec = {
-				title = util.striptags(subnode.title),
+				title = xml.striptags(subnode.title),
 				order = subnode.order
 			}
 
@@ -741,7 +742,7 @@ local function init_template_engine(ctx)
 				(scope and type(scope[key]) ~= "function" and scope[key]) or "")
 
 			if noescape ~= true then
-				val = util.pcdata(val)
+				val = xml.pcdata(val)
 			end
 
 			return string.format(' %s="%s"', tostring(key), val)
@@ -756,8 +757,8 @@ local function init_template_engine(ctx)
 		translate   = i18n.translate;
 		translatef  = i18n.translatef;
 		export      = function(k, v) if tpl.context.viewns[k] == nil then tpl.context.viewns[k] = v end end;
-		striptags   = util.striptags;
-		pcdata      = util.pcdata;
+		striptags   = xml.striptags;
+		pcdata      = xml.pcdata;
 		media       = media;
 		theme       = fs.basename(media);
 		resource    = luci.config.main.resourcebase;
diff --git a/modules/luci-mod-system/luasrc/model/cbi/admin_system/backupfiles.lua b/modules/luci-mod-system/luasrc/model/cbi/admin_system/backupfiles.lua
index c81466c87..91c654916 100644
--- a/modules/luci-mod-system/luasrc/model/cbi/admin_system/backupfiles.lua
+++ b/modules/luci-mod-system/luasrc/model/cbi/admin_system/backupfiles.lua
@@ -68,7 +68,7 @@ else
 					break
 				else
 					files[#files+1] = "<li>"
-					files[#files+1] = luci.util.pcdata(ln)
+					files[#files+1] = luci.xml.pcdata(ln)
 					files[#files+1] = "</li>"
 				end
 			end
-- 
2.20.1


From 83ad2f793c4e23e67e112c63e7c643c6683a0038 Mon Sep 17 00:00:00 2001
From: Sven Roederer <freifunk@it-solutions.geroedel.de>
Date: Sat, 6 Jul 2019 14:31:15 +0200
Subject: [PATCH 5/5] luci-base: update dependency to luci-lib-base

To satisfy the dependencies.

Signed-off-by: Sven Roederer <freifunk@it-solutions.geroedel.de>
---
 modules/luci-base/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/luci-base/Makefile b/modules/luci-base/Makefile
index 501964661..6cb2b6409 100644
--- a/modules/luci-base/Makefile
+++ b/modules/luci-base/Makefile
@@ -12,7 +12,7 @@ LUCI_TYPE:=mod
 LUCI_BASENAME:=base
 
 LUCI_TITLE:=LuCI core libraries
-LUCI_DEPENDS:=+lua +luci-lib-nixio +luci-lib-ip +rpcd +libubus-lua +luci-lib-jsonc +liblucihttp-lua +rpcd-mod-file +rpcd-mod-luci +cgi-io
+LUCI_DEPENDS:=+lua +luci-lib-nixio +luci-lib-ip +rpcd +libubus-lua +luci-lib-jsonc +liblucihttp-lua +luci-lib-base +rpcd-mod-file +rpcd-mod-luci +cgi-io
 
 PKG_LICENSE:=MIT
 
-- 
2.20.1

